src/app/content/components/NudgeStudyTools/utils.spec.tsx

```diff
@@ -0,0 +1,247 @@
+  });
+
+  it('passedTimeInterval - return true if passed allowed time interval or last date is undefined', () => {
+    expect(utils.passedTimeInterval()).toEqual(true);
+    const now = new Date();
+    const passed = new Date(now.getTime() - constants.timeIntervalBetweenShowingNudgeInMs - 10000);
+    Cookies.set(constants.cookieNudge.date, passed.toString());
+    expect(utils.passedTimeInterval()).toEqual(true);
+  });
+
+  it('incrementPageCounterCookie increments page counter cookie', () => {
+    expect(utils.getPageCounterCookie()).toEqual(0);
+    utils.incrementPageCounterCookie();
+    utils.incrementPageCounterCookie();
+    expect(utils.getPageCounterCookie()).toEqual(2);
+  });
+
+  it('shouldDisplayNudgeStudyTools - return true only if all criteria passes', () => {
+    expect(utils.shouldDisplayNudgeStudyTools()).toEqual(false);
+
+    const now = new Date();
+    const passed = new Date(now.getTime() - constants.timeIntervalBetweenShowingNudgeInMs - 10000);
+    Cookies.set(constants.cookieNudge.date, passed.toString());
+    expect(utils.shouldDisplayNudgeStudyTools()).toEqual(false);
+
+    Cookies.set(constants.cookieNudge.pageCounter, constants.nudgeStudyToolsMinPageLimit.toString());
+    expect(utils.shouldDisplayNudgeStudyTools()).toEqual(true);
+
+    Cookies.set(constants.cookieNudge.counter, constants.nudgeStudyToolsShowLimit.toString());
+    expect(utils.shouldDisplayNudgeStudyTools()).toEqual(false);
+  });
+
+  it('setNudgeStudyToolsCookies', () => {
+    Cookies.set(constants.cookieNudge.pageCounter, '1');
+    const counter = utils.getCounterCookie();
+    const pageCounter = utils.getPageCounterCookie();
+    const date = utils.getDateCookie();
+    expect(counter).toEqual(0);
+    expect(pageCounter).toEqual(1);
+    expect(date).toEqual(undefined);
+
+    utils.setNudgeStudyToolsCookies();
+
+    expect(utils.getCounterCookie()).toEqual(1);
+    expect(utils.getPageCounterCookie()).toEqual(0);
+    expect(utils.getDateCookie()).toBeTruthy();
+  });
+
+  it('useIncrementPageCounter - increments page counter cookie to until it reaches limit', () => {
+    const store = createTestStore();
+    // tslint:disable-next-line: variable-name
+    const Component = () => {
+      utils.useIncrementPageCounter();
+      return <div/>;
+    };
+
+    renderer.create(<Provider store={store}>
+      <Component />
+    </Provider>);
+
+    renderer.act(() => {
+      expect(utils.getPageCounterCookie()).toEqual(0);
+      store.dispatch(receivePage({...shortPage, references: []}));
+    });
+
+    renderer.act(() => {
+      expect(utils.getPageCounterCookie()).toEqual(1);
+      store.dispatch(receivePage({...shortPage, references: []}));
+    });
+
+    renderer.act(() => {
+      expect(utils.getPageCounterCookie()).toEqual(2);
+      store.dispatch(receivePage({...shortPage, references: []}));
+    });
+
+    renderer.act(() => {
+      expect(utils.getPageCounterCookie()).toEqual(3);
+    });
```

Comment:
I feel like those assertions are better off outside of `renderer.act`. It's meant to prepare components for assertions, not contain any by itself.